src/python/foglamp/data_purge/sqlalchemy_purge.py

```diff
@@ -0,0 +1,324 @@
+        else:
+            time_dict[value] = tmp
+
+    time_in_sec = datetime.timedelta(seconds=0)
+    time_in_min = datetime.timedelta(minutes=0)
+    time_in_hr = datetime.timedelta(hours=0)
+    time_in_day = datetime.timedelta(days=0)
+
+    for key in time_dict.keys():
+        if 'sec' in key:
+            time_in_sec = datetime.timedelta(seconds=time_dict[key])
+        elif 'min' in key:
+            time_in_min = datetime.timedelta(minutes=time_dict[key])
+        elif ('hr' in key) or ('hour' in key):
+            time_in_hr = datetime.timedelta(hours=time_dict[key])
+        elif ('day' in key) or ('dy' in key):
+            time_in_day = datetime.timedelta(days=time_dict[key])
+    return time_in_sec+time_in_min+time_in_hr+time_in_day
+
+
+def convert_sleep(set_time: str) -> int:
+    """Convert "wait" in config file to seconds in order to know how long to wait until next purge process. 
+        This method would potentially be replaced by the scheduler 
+    Args:
+        set_time (str): A string of "values" specified in the config to declare how long to wait till next purge
+
+    Returns:
+        integer (of seconds) based on wait in config 
+    """
+    if set_time.isdigit():
+        return int(set_time)
+    time_dict = {}
+    tmp = 0
+    for value in set_time.split(" "):
+        if value.isdigit() is True:
+            tmp = int(value)
+        else:
+            time_dict[value] = tmp
+    time_in_sec = 0
+    time_in_min = 0
+    time_in_hr = 0
+    time_in_dy = 0
+    for key in time_dict:
+        if "sec" in key:
+            time_in_sec = time_dict[key]
+        elif "min" in key:
+            time_in_min = 60 * time_dict[key]
+        elif ("hour" in key) or ("hr" in key):
+            time_in_hr = 60 * 60 * time_dict[key]
+        elif ("day" in key) or ("dy" in key):
+            time_in_dy = 60 * 60 * 24 * time_dict[key]
+        else:
+            print("Error: Invalid Value(s) in config file")
+            sys.exit()
+    return time_in_sec+time_in_min+time_in_hr+time_in_dy
+
+
+def get_count(stmt: str) -> int:
+    """Imitate connection to postgres that returns result.
+    
+    Since all queries return either an ID (integer) or COUNT (integer) the expected 
+    return value is also an integer. 
+    
+    Args:
+        stmt (str): generated SQL query   
+    Returns:
+        Integer value result of SQL query
+    """
+    try:
+        query_result = conn.execute(stmt)
+    except conn.Error as e:
+        print(e)
+        sys.exit()
+    else:
+        return int(query_result.fetchall()[0][0])
+
+
+def execute_delete(stmt: str) -> None:
+    """Imitate connection to Postgres and execute DELETE query
+    Args:
+        stmt (str): DELETE stmt 
+    Returns: 
+        The method doesn't return anything
+    """
+
+    try:
+        conn.execute(stmt)
+    except conn.Error as e:
+        print(e)
+        sys.exit()
+    else:
+        conn.execute("commit")
+
+
+def get_nth_id() -> None:
```

Comment:
Yes indeed, when the purge process uses the stream table this will not be needed